core/src/test/java/com/sap/ai/sdk/core/common/ClientStreamingHandlerTest.java

```diff
@@ -0,0 +1,110 @@
+package com.sap.ai.sdk.core.common;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParseException;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.Data;
+import lombok.SneakyThrows;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.http.message.BasicClassicHttpResponse;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for {@link ClientStreamingHandler}. Inherits common test utilities and mock classes from
+ * ClientResponseHandlerTest.
+ */
+class ClientStreamingHandlerTest extends ClientResponseHandlerTest {
+
+  @Data
+  static class MyStreamedDelta implements StreamedDelta {
+    @JsonProperty("value")
+    private String value; // Simulates the content
+
+    @JsonProperty("finish_reason")
+    private String finishReason;
+
+    @Nonnull
+    @Override
+    public String getDeltaContent() {
+      return value != null ? value : "";
+    }
+
+    @Nullable
+    @Override
+    public String getFinishReason() {
+      return finishReason;
+    }
+  }
+
+  @SneakyThrows
+  @Test
+  void testHandleStreamingResponse_variousScenarios() {
```

Comment:
```suggestion
  void testHandleStreamingResponse() {
```
camel case + snake case?